bundles/org.openhab.binding.argoclima/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,495 @@
+				<description><![CDATA[If enabled, will directly communicate with the device to send commands and poll for its status.<br/>
+				Must be enabled for <code>LOCAL_CONNECTION</code> mode. If disabled (in any of the <code>REMOTE_*</code> connection modes), no push communication will be initiated from OH-side, and commands will be sent on next device-side poll cycle (possible delay)]]></description>
+				<default>true</default>
+				<advanced>true</advanced>
+			</parameter>
+
+			<parameter name="refreshInterval" type="integer" unit="s" min="0" groupName="connection" required="false">
+				<label>Refresh Interval</label>
+				<description><![CDATA[Interval the device is polled in seconds. This setting is only effective if <code>Use Local Connection</code> is <code>ON</code>.<br/>
+				Set to <code>0</code> to disable polling (requires any of <code>REMOTE_*</code> connection modes)]]></description>
+				<default>30</default>
+				<advanced>true</advanced>
+			</parameter>
+
+			<parameter name="stubServerPort" type="integer" min="0" max="65536" required="false" groupName="serverStub">
+				<label>Stub server listen port</label>
+				<description><![CDATA[Port at which the Stub server will listen on.<br/>
+				<i>This setting is effective only for <code>REMOTE_*</code> connection modes</i>
+				]]>
+				</description>
+				<default>8239</default>
+				<advanced>true</advanced>
+			</parameter>
+
+			<parameter name="stubServerListenAddresses" type="text" multiple="true" required="false"
+				groupName="serverStub">
+				<context>network-address</context>
+				<label>Stub server listen interfaces</label>
+				<description><![CDATA[List of interfaces the stub server will listen on<br/>
+				<i>This setting is effective only for <code>REMOTE_API_STUB | REMOTE_API_PROXY</code> connection modes</i>
+				]]>
+				</description>
+				<default>0.0.0.0</default>
+				<advanced>true</advanced>
+			</parameter>
+
+			<parameter name="oemServerAddress" type="text" required="false" groupName="serverStub">
+				<context>network-address</context>
+				<label>Argo remote server address</label>
+				<description><![CDATA[The OEM server's port, used to pass through the communications to. <br/>
+				<i>This setting is effective only for <code>REMOTE_API_PROXY</code> connection mode</i>
+				]]>
+				</description>
+				<default>31.14.128.210</default>
+				<advanced>true</advanced>
+			</parameter>
+
+			<parameter name="oemServerPort" type="integer" min="0" max="65536" required="false" groupName="serverStub">
+				<label>Argo remote server port</label>
+				<description><![CDATA[The OEM server's port, used to pass through the communications to<br/>
+				<i>This setting is effective only for <code>REMOTE_API_PROXY</code> connection mode</i>
+				]]>
+				</description>
+				<default>80</default>
+				<advanced>true</advanced>
+			</parameter>
+
+			<parameter name="showCleartextPasswords" type="boolean" required="false" groupName="serverStub">
```

Comment:
Well, this is **very** unique to this device (and has nothing to do with user-provided user-password). The ["note on security"](https://github.com/mbronk/openhab-addons/blob/mbronk/argoclima_binding_wip_OH4.1-SNAPSHOT-clean/bundles/org.openhab.binding.argoclima/README.md#argo-protocol-details) hopefully explains it a bit...

All in all, the device happily exfiltrates passwords to itself and your home wifi (these are not the ones you enter into OpenHab, but device-side!). Reason I added this switch is... awareness (noone reads README's right? :)).  And I cannot rely on browser/framework, since these are dynamic **device properties**, not user password input.

I was actually considering scrambling these passwords (in a pass-through mode), so that the vendor's cloud doesn't learn your WiFi password, but I have no clue if this won't create any undefined behavior device-side, and don't have time to do a more extensive reverse-engineering of the device firmware... so left as a "hide" option, not really doing much rather than masking properties.

I can delete this if you prefer (as I agree it's not really adding meaningful security), though TBH I'd prefer to keep this concept (for awareness)

Reply:
Ah, i see now. This property is read only and originated from the device. 
What is the benefit of exposing the password here? If there is no specific reason, i would opt to completely remove the property and store it elsewhere. It may also end up in logging, screenshots etc.

Reply:
> This property is read only and originated from the device.

Yes

> What is the benefit of exposing the password here?

Awareness! :) 
It's not like when I stop displaying these, the device will magically stop sending the passwords. I'd rather much prefer everyone seeing their Wi-Fi password "magically" appearing in openHAB, than pretending this is secure.
BTW, this also gives you the device login/password (which are NOT changeable)... so it can be thought as a... password recovery technique of sorts? :)


I know it seems like a YAGNI, but short of submitting a CVE about the device, I thought it is the least I could do... and my conscience couldn't bear with just putting **plain** password as property by default, so I went with this "_default-masked, opt-in displayed_" approach :)
Bottom line, I'm happy to adjust to be more "bearable", but would really like to keep this concept :)

Reply:
I would advice to remove the password property as it has no function to openHAB or the user. I don't think showing the password (masked or not) in openHAB is the right way of creating awareness, in practice it will only lead to another possible place to leak the secret.

Reply:
I see your point. At the same time would like to retain this as an option (as I find the `device password` display useful (in fact, it may be the only way to see the device-assigned password in a stub mode).
So how about a middle ground: 
- convert the parameter to an enum: `includeDeviceSidePasswordsInProperties: NEVER | MASKED | CLEARTEXT` ?

Would it work for you?

Reply:
Ok, that is something to work with. Defaults to never, or am I pushing too much 😂

Reply:
> Defaults to never, or am I pushing too much 😂

Naah... you got yourself a deal! :-)
Will try to update shortly. Meanwhile feel free to review other changes.

Reply:
Updated as per this discussion, please take a look.